internal/storage/caching/fast_stat_bucket.go

```diff
@@ -204,8 +205,10 @@ func (b *fastStatBucket) StatObject(
 			return
 		}
 
-		// Otherwise, return the object.
-		o = entry
+		// Otherwise, return object and ExtendedObjectAttributes.
+		minObj := storageutil.ConvertObjToMinObject(entry)
+		m = &minObj
+		e = storageutil.ConvertObjToExtendedAttributes(entry)
```

Comment:
we can simply return nil for ExtendedObjectAttributes when cache is hit ?

Reply:
FYI: Handling ReturnExtendedObjectAttributes parameter in the stat object request was essential for this change to work as other unit tests started failing. So this PR now includes updates to `fast_stat_bucket.go` and `bucket_handle.go` to properly implement StatObject method.